### PR TITLE
Improves Curator display cases

### PR DESCRIPTION
**Pull request by cacogen**
:cl: cacogen
add: Items saved in Curator display cases can now be removed (roundstart items are still non-functional replicas though)
tweak: You can now open Curator display cases with the key to get items out of them
tweak: Curator display cases no longer eat items they're clicked with, they have to be open and empty to put things in them
:cl:

[why]: # (Please add a short description [on the next line] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding:)
I rolled Curator for the first time today and I was a little disappointed by the display cases. The way they worked wasn't very intuitive, and they had a habit of eating items and then had to be destroyed to get them back.

I really like the item persistence thing, but I wanted to actually be able to get the items out and play with them and I didn't like having to delete them to display something of my own.

This change makes it so that the non-functional replicas saved in the cases can be taken out. They have inhand sprites now so it actually looks like you're carrying the item and their specific names are now preserved from the previous round so that unique items such as IDs and PDAs keep their names.

Bear in mind these are just /obj/items with the sprite of the object that was saved, so they have no functionality whatsoever.

Other stuff this pull does:

* You can add and remove displayed items from cases as much as you want
* The key now has to be in your active hand to open and close the case, so you don't accidentally toggle the lock when trying to display something
* Using an object on the case while it's closed results in you attacking the case with it instead of trying to display it
* Fixed some of the pronouns in user messages so they look better
* Sets the name and description of replicas to indicate that they're replicas of the item and not the item itself

Known issues:

* ~~Existing saved trophies didn't have their names saved, so this may cause a runtime at roundstart when spawning them (not sure though, if someone could confirm this then I'll add a check for name before reading from it)~~ added a check just in case
* ~~Each time a trophy is added to a case, it's saved. This means the pool of saved trophies can be spammed with garbage. I'd like to make it so when a trophy is removed from a case it's removed from the pool, but I'm not sure how to do that.~~ can only save a trophy once per case per round again
* Overlays aren't preserved (e.g. gun charge meters and reagents in beakers). I don't think this can be fixed without a ton of work or just giving them the item itself (which could be possible if you assigned a cost to each item spawned, but would take a ton of work because you'd then also have to create a system that decides which variables to save based on object type).
* Unique inhands (e.g. gun sprites with visible charge meters) don't show up at all. I might fix this later on because otherwise it kind of defeats the purpose of having gun replicas for example.

### CODE REVIEW

**Comment by Steelpoint** [comment]
If you put a item in, and remove it during the same round, will it be plastic or will it be functional?

Otherwise I think this is a interesting change.

**Comment by cacogen** [comment]
Functional

**Comment by improvedname** [comment]
The ultimate metabreaker

**Comment by Profakos** [comment]
Ah, you beat me to it, and did it better than I would have done.
But my suggestion is that only the first new item inserted is saved, further insertion could say "Error, historical buffers are full" or something akin to it. But then again, that would result in someone putting in an important item to Persist, and then taking it out and using it nonchalantly.

Other possibility is that instead of placing the item in the case, a trophy case saves items at roundend, or when it completely breaks.

**Comment by improvedname** [comment]
It also be cool if you could vote so you can how much the crew likes your shit

**Comment by fludd12** [comment]
Something to note, if you make it so that removing a plastic replica from a case removes it from the item pool, would re-adding it do the opposite, or add a plastic replica of a plastic replica?

**Comment by cacogen** [comment]
@Profakos 
I currently have it so when they insert an item into a case they can choose to save it if they want, and they can only do this once per case per round. Haven't updated the pull with that code yet though.

I don't think there's anything wrong with saving an item and then taking it out and using it.

I kind of wanted to make it so that what's in the trophy case at the end of the round gets saved, but I'm not sure how to do that (or if I'd be allowed to do that).

@improvedname 
Yeah I want to add that actually. So that there's some quality control as well. Similar to engravings. But if it's downvoted enough it's removed from the pool. I'd have to figure out how to remove things from the pool though.

@fludd12 
Yeah you can save a replica, so you can do this endlessly if you want. I guess I should stop this from being possible but I kind of liked the idea of it.

**Comment by Profakos** [comment]
Secret satchels and engravings are polled at roundend, and since there is a global list of trophy cases, you can do something similar to save the contents as long as it is not a replica. I intended to do something like this originally, but then I realized that you would have to keep on defending the cases.
Making them also save their content when they get deconstruct()-ed could solve this issue though.

**Comment by RandomMarine** [comment]
I didn't like the pool from the beginning. It needs some way to clean out garbage entries because it's flooded with librarian PDAs, bookbags, and ids.

File: code/game/objects/structures/displaycase.dm [hunk 1]
```diff
@@ -331,33 +320,40 @@
 			else
 				to_chat(user, "You are too far to set the plaque's text.")
 
-		SSpersistence.SaveTrophy(src)
+		if(scanned_for_replication == FALSE)
+			var/scan = alert(user, "Scan this item for replication in future shifts? You can only do this once.", "Scan for replication?", "No", "Yes")
```
Comment (AnturK) [inline code/game/objects/structures/displaycase.dm]: You can keep this open to fill the saved list multiple times

Comment (cacogen) [inline code/game/objects/structures/displaycase.dm]: so how do I fix that?

**Comment by cacogen** [comment]
@RandomMarine 

I'm going to try copying the voting system from engravings and make it so anything below say -5 points will automatically be removed. It'd come with the Delete button admins get for engravings as well.

File: code/game/objects/structures/displaycase.dm (hunk 1 above)
Comment (AnturK) [inline code/game/objects/structures/displaycase.dm]: Add a variable that checks if the display is currently in use and stop the interaction when it's set. Set it here or at the start of the whole chain.

**Comment by lzimann** [comment]
Still doing this?

**Comment by lzimann** [comment]
Ping if you want this reopen.

**Pull request closed without merging**